main-service/src/main/java/ru/practicum/ewm/mainservice/comment/service/CommentServiceImpl.java

```diff
@@ -0,0 +1,185 @@
+package ru.practicum.ewm.mainservice.comment.service;
+
+import com.querydsl.core.BooleanBuilder;
+import com.querydsl.core.types.OrderSpecifier;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ru.practicum.ewm.mainservice.comment.dto.CommentDto;
+import ru.practicum.ewm.mainservice.comment.dto.CommentQueryParams;
+import ru.practicum.ewm.mainservice.comment.dto.NewCommentDto;
+import ru.practicum.ewm.mainservice.comment.mapper.CommentMapper;
+import ru.practicum.ewm.mainservice.comment.model.Comment;
+import ru.practicum.ewm.mainservice.comment.model.QComment;
+import ru.practicum.ewm.mainservice.comment.repository.CommentRepository;
+import ru.practicum.ewm.mainservice.event.model.Event;
+import ru.practicum.ewm.mainservice.event.model.state.EventState;
+import ru.practicum.ewm.mainservice.event.repository.EventRepository;
+import ru.practicum.ewm.mainservice.exception.BadRequestException;
+import ru.practicum.ewm.mainservice.exception.ConflictException;
+import ru.practicum.ewm.mainservice.exception.NotFoundException;
+import ru.practicum.ewm.mainservice.user.model.User;
+import ru.practicum.ewm.mainservice.user.repository.UserRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class CommentServiceImpl implements CommentService {
+    private final CommentRepository commentRepository;
+    private final CommentMapper commentMapper;
```

Comment:
Мапперы чаще всего содержат (или должны содержать статические методы и нет необходимости декларировать их, как поля класса)

Reply:
Не до конца понял, что имелось в виду. В проекте используется `MapStruct`, а `CommentMapper` является интерфейсом. Можно вручную получить инстанс с помощью метода `Mappers.getMapper()`, тогда не придется декларировать как поле класса, но [более предпочтительной альтернативой является внедрение через DI](https://www.baeldung.com/mapstruct):

> Next, let’s obtain an instance of a mapper in MapStruct by merely calling Mappers.getMapper(YourClass.class).
>Of course, that’s a very manual way of getting the instance. However, a much better alternative is to inject the mapper directly where we need it (if our project uses any Dependency Injection solution).
>Luckily, MapStruct has solid support for both Spring and CDI (Contexts and Dependency Injection).
>To use Spring IoC in our mapper, we need to add the componentModel attribute to @Mapper with the value spring, and for CDI, it would be cdi.